packages/material-ui/src/utils/setRef.ts

```diff
@@ -11,6 +15,12 @@
  * @param ref a ref callback or ref object if anything falsy this is a no-op
  */
 export default function setRef<T>(
-  ref: React.RefObject<T> | ((instance: T | null) => void) | null | undefined,
-  value: T | null
-): void;
+  ref: React.MutableRefObject<T | null> | ((instance: T | null) => void) | null | undefined,
+  value: T | null,
+) {
```

Comment:
```suggestion
): void {
```

For modules we want to stick with explicit return types for readability.